pulser-core/pulser/sequence/_schedule.py

```diff
@@ -94,19 +94,32 @@ def get_samples(self, modulated: bool = False) -> ChannelSamples:
             amp[s.ti : s.tf] += pulse.amplitude.samples
             det[s.ti : s.tf] += pulse.detuning.samples
             ph_jump_t = self.channel_obj.phase_jump_time
-            t_start = max(0, (s.ti - ph_jump_t))
+            t_start = s.ti - ph_jump_t if ind > 0 else 0
             t_end = (
                 channel_slots[ind + 1].ti - ph_jump_t
                 if ind < len(channel_slots) - 1
                 else dt
             )
             phase[t_start:t_end] += pulse.phase
-            slots.append(_TargetSlot(s.ti, s.tf, s.targets))
+            tf = s.tf
+            if modulated:
+                # Account for as much fall time as possible
+                fall_time = pulse.fall_time(self.channel_obj)
+                tf += (
+                    min(fall_time, channel_slots[ind + 1].ti - s.tf)
+                    if ind < len(channel_slots) - 1
+                    else fall_time
+                )
```

Comment:
I'm sorry about that, I'll add a comment. 
When there is modulation, the pulse gets elongated - this increase in duration is the `fall_time`, the extra time taken to go back to zero. If two pulses are adjacent or have a space between them that is smaller than the fall time, then they will be merged instead (ie the first one doesn't get to go back to zero before the second starts). 
The reason we need to account for this in the target slots is because they set the limits when a pulse starts and ends, which is used when separating the samples between targets in the nested dict. If we leave the target slots untouched, the modulated pulses will be truncated.

Reply:
Thanks it's much clearer, I was aware of this effect but it was unclear that was the reason behind these lines